Content.Shared/Silicons/StationAi/SharedStationAiSystem.cs

```diff
@@ -226,6 +241,44 @@ private void OnHolderInteract(Entity<StationAiHolderComponent> ent, ref AfterInt
         }
     }
 
+    private void OnHolderInteract(Entity<StationAiHolderComponent> ent, ref AfterInteractEvent args)
+    {
+        if(!args.CanReach)
+            return;
+        
+        if (!TryComp(args.Target, out StationAiHolderComponent? targetHolder))
+            return;
+        
+        if (!TryComp(args.Used, out IntellicardComponent? intelliComp))
+            return;
+        
+        bool isUploading = _slots.CanEject(ent.Owner, args.User, ent.Comp.Slot);
+        
+        bool isDownloading = _slots.CanEject(args.Target.Value, args.User, targetHolder.Slot);
+        
+        if(isUploading == isDownloading)
+            return;
+        
+        if(TryGetHeldFromHolder((targetHolder.Owner, targetHolder), out var held) && _timing.CurTime > intelliComp.NextWarningAllowed) {
```

Comment:
```suggestion
        if (TryGetHeldFromHolder((args.Target, targetHolder), out var held) && _timing.CurTime > intelliComp.NextWarningAllowed) {
```
Using component.Owner is a big no! That is obsolete
![grafik](https://github.com/user-attachments/assets/56daf2bc-8b49-44a1-874c-a6a23fe438bf)
you also already got the owning entity, it's where you got `targetHolder` from in the first place